Update GoogleSignIn to 6.0.1

dvdchr opened the pull request:
Refs https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/issues/609

This change updates `WordPressAuthenticator` dependency to `1.41.0-beta.2`, which upgrades `GoogleSignIn` to version `6.0.1`. Note that https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/610 needs to be merged first before merging this PR.

If this is too close to 7.3, feel free to let me know and I'll bump the milestone to 7.4 🙂 .

## To Test

- Ensure that the project compiles successfully.
- Ensure that Google sign-in flows work as expected. 

## Author Checklist

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

dvdchr committed:
Update authenticator to 1.41.0-beta.2

pmusolino reviewed (approved):
The authentication worked as expected :shipit: thanks for updating it @dvdchr.

ealeksandrov reviewed (approved):
Integration works well, tested as part of https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/610 :shipit:

Couple of side notes:
- specifying version with `1.41.0-beta` would work better, supporting all betas for 1.41 automatically without need to switch branch names (assuming we have `1.41.0-beta.2` tag for this change in WPAuth)
- I think this one deserves `[internal]` [release notes](https://github.com/woocommerce/woocommerce-ios/blob/develop/RELEASE-NOTES.txt) mention for QA

dvdchr committed:
Update release notes

dvdchr committed:
Update WordPressAuthenticator dependency to 1.41.0-beta

dvdchr committed:
Merge branch 'develop' into feature/authenticator-googleSignIn-6.0

dvdchr committed:
minor: fix indentation

dvdchr committed:
Update Podfile.lock

dvdchr committed:
minor: remove extra space

dvdchr committed:
Update Podfile.lock

dvdchr commented:
Thanks @ealeksandrov!

> specifying version with 1.41.0-beta would work better, supporting all betas for 1.41 automatically without need to switch branch names (assuming we have 1.41.0-beta.2 tag for this change in WPAuth)

addressed in https://github.com/woocommerce/woocommerce-ios/pull/4725/commits/7240b279ee216662b1658334d1e611984d415d85.

> I think this one deserves [internal] release notes mention for QA

addressed in https://github.com/woocommerce/woocommerce-ios/pull/4725/commits/496d29dbcb969da0625eae6c6d57b664cab350ac.
